plugins/TrollManagement/class.trollmanagement.plugin.php

```diff
@@ -469,4 +499,189 @@ private function checkTroll(int $userID, array &$args) {
+        if (!is_null($fingerprint)) {
+            $sql = clone Gdn::sql();
+            $sql->reset();
+            $users = $sql
+                ->select('userID AS siblingsCount', 'count')
+                ->from('User')
+                ->where('Fingerprint', $fingerprint)
+                ->get()->firstRow(DATASET_TYPE_ARRAY);
+            return $users['siblingsCount'];
+        }
+        return 0;
+    }
+
+    /**
+     * Upon user registration, we trigger an early Fingerprint tag & we check if this new registration trips the
+     * maximum amount of user accounts allowed for a single fingerprint. If it is, we give this new user the
+     * 'Applicant' role.
+     *
+     * @param UserModel $sender
+     * @param array $args
+     */
+    public function userModel_afterRegister_handler($sender, $args) {
+        if (c('TrollManagement.PerFingerPrint.Enabled', false) == true) {
+            $userID = $args['UserID'];
+
+            $maxSiblingAccounts = c('TrollManagement.PerFingerPrint.MaxUserAccounts');
+            $userFingerprint = $this->setFingerprint($userID);
+
+            if ($userFingerprint !== false) {
+                $fingerprintUsages = $this->getSharedFingerprintsUsersCount($userFingerprint);
+                if ($fingerprintUsages >= $maxSiblingAccounts) {
+                    Gdn::userModel()->addRoles($userID, [RoleModel::APPLICANT_ID], true);
+                }
+            }
+        }
+    }
+
+    /**
+     * Adds "Fingerprint" to the allowed banTypes of the "Ban Rules" dashboard interface.
+     *
+     * @param array $banTypes
+     * @return array
+     */
+    public function settingsController_listBanTypes(array $banTypes): array {
+        $banTypes['Fingerprint'] = t('Fingerprint');
+        return $banTypes;
+    }
+
+    /**
+     * Either add "Fingerprint" row header or an fingerprint value to the dashboard's list of users.
+     *
+     * @param Controller $sender
+     * @param array $args
+     */
+    public function base_userCell_handler($sender, $args) {
+        // If we have user data, we create a cell containing the fingerprint.
+        if (isset($args['User'])) {
+            echo '<td>' . val('Fingerprint', $args['User']) . '</td>';
+        } else {
+            // Otherwise, we output a column header.
+            $get = Gdn::request()->get();
+            $get['order'] = 'Fingerprint';
+            $orderUrl = '/dashboard/user?'.http_build_query($get);
+
+            echo '<th class="column-md">' . anchor(t('Fingerprint'), $orderUrl) . '</th>';
+        }
+    }
+
+    /**
+     * Add "Fingerprint" to the possible ban query.
+     *
+     * @param array $result
+     * @param array $ban
+     * @return array
+     */
+    public function banModel_banWhere_handler(array $result, array $ban): array {
+        switch (strtolower($ban['BanType'])) {
+            case 'fingerprint':
+                $result['u.Fingerprint like'] = $ban['BanValue'];
```

Comment:
I don't think a "like" is useful here. Aren't these values more or less random? Why would you want to ban based on a partial match?